Core/Src/main.c

```diff
@@ -65,6 +68,13 @@ void SystemClock_Config(void);
 int main(void)
 {
   /* USER CODE BEGIN 1 */
+	uint8_t spi_rx_buf[3]; //receive buffer
+	uint8_t spi_tx_buf[3]; //transmit buffer
+	uint16_t adc_val = 0x0000;
+	uint16_t MAX_ADC_VAL = 1023; //equals 2^10 - 1, shouldn't this be higher than MIN_COMPARE_VAL?
```

Comment:
This is correct, the compare value is for the timer. Once you have the basically percent of the MAX_ADC_VAL that is being read, you will scale that for the timer compare.